makeStage2.sh

```diff
@@ -39,5 +40,66 @@ cd $usepop/pop/x/src/
 # ways to construct the Poplog system.
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
+# $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p ${BUILD_HOME}/environments
+
+# We need to capture the Poplog environment for each build variant.
```

Comment:
Can we get a bit more information here. Can you give a few examples of variables that are needed, and also which variables need to be filtered out and why. I see you're filtering out `SHLVL` and `PWD` (both of which make sense), but why the other `pop*` vars?

Reply:
Can you also document what the args to the function are. Here I suppose it's a value for `$usepop`?

Reply:
Why do we need to capture the environment and who is going to use it/for what purpose?

Reply:
Partially answered. The logic for filtering out the other variables I haven't done yet.